src/Microsoft.AspNetCore.Mvc.RazorPages/Internal/DefaultPageHandlerMethodSelector.cs

```diff
@@ -65,6 +65,12 @@ private static List<HandlerMethodDescriptor> SelectHandlers(PageContext context)
             List<HandlerMethodDescriptor> handlersToConsider = null;
 
             var formAction = Convert.ToString(context.RouteData.Values[FormAction]);
+
+            if (string.IsNullOrEmpty(formAction))
+            {
+                formAction = Convert.ToString(context.HttpContext.Request.Query[FormAction]);
```

Comment:
It returns `StringValues`. Is the case where this would matter `?formaction=Whatever&formaction=AnotherThing`.